gobblin-temporal/src/main/java/org/apache/gobblin/temporal/ddm/workflow/impl/ExecuteGobblinWorkflowImpl.java

```diff
@@ -210,6 +211,7 @@ protected static Set<String> calculateWorkDirsToDelete(String jobId, Set<String>
         resultSet.add(dir.toString());
       } else {
         log.warn("Skipping deletion of directory {} as it is not associated with job {}", dir, jobId);
+        throw new IOException("Not all work directories are contained within the current job execution, please validate the staging and output directories");
```

Comment:
please put the specific info from the log above here too, as that's what shows up in the temporal UI:
```
log.warn("Skipping deletion of directory {} as it is not associated with job {}", dir, jobId);
```